### DISCUSSION TITLE
Gateway API RouteResource for kafka?

### DISCUSSION THREAD

**Discussion by jcpunk** [strimzi/strimzi-kafka-operator#8399] [Ideas]
With the new gateway api, would it be possible for someone to build a kafka route resource to help make integration with external clients easier?

https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-resources

I'll confess I know little to nothing about either side of the problem space here, but was hopeful this might be a helpful question.

**Comment by scholzj** [comment]
You can currently set it up manually today if you want by creating the TLSRoute or TCPRoute resources. I'm not sure if / when we will add support for it directly into the operator. The Gateway API is still in development so it is not completely clear what changes will come in the next versions and compatibility might break. I'm also curious to see how compatible the different implementations will be.

I planned to write a blog post about it - if it is not urgent for you, maybe you can wait for it. I can try to get to it next weekend or something like that.

Note: I used the Envoy Gateway for my tests and that worked fine -> not every Gateway API implementation supports the TLSRoute or TCPRoute resources needed for Kafka. So the usability also depends on what gateway implementation you plan to use.

**Reply by jcpunk** [reply]
This was a far in the future thought, so I can certainly wait until you've got time for the blog post.

**Reply by scholzj** [reply]
> I planned to write a blog post about it - if it is not urgent for you, maybe you can wait for it. I can try to get to it next weekend or something like that.

Sorry, a week is very long on my planet ... so the next weekend did not happen yet :-/. But I still plan to get to it.
